stdlib/src/os/atomic.mojo

```diff
@@ -24,7 +24,7 @@ from memory import UnsafePointer, bitcast
 from sys.info import triple_is_nvidia_cuda
 
 
-struct Atomic[type: DType]:
+struct Atomic[type: DType](ExplicitlyCopyable):
```

Comment:
This was to allow me to construct the [0|1] length array for weak. I wanted to revisit the specifics, since having an array of atomics can be useful in a lot of contexts but there isn't currently a good way of constructing one without arcane MaybeUninit incantations. I do have some questions about the atomic type that can be had out-of-thread too, that (I think) make Arc technically unsound in a single threaded context and relatively impossible to use in a multi-threaded one